bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/internal/handler/GenericMQTTThingHandler.java

```diff
@@ -144,6 +161,29 @@ public void initialize() {
                 continue;
             }
             final ChannelConfig channelConfig = channel.getConfiguration().as(ChannelConfig.class);
+
+            if (channelTypeUID
+                    .equals(new ChannelTypeUID(MqttBindingConstants.BINDING_ID, MqttBindingConstants.NUMBER))) {
+                Unit<?> unit = UnitUtils.parseUnit(channelConfig.unit);
+                String dimension = unit == null ? null : UnitUtils.getDimensionName(unit);
+                String expectedItemType = dimension == null ? "Number" : "Number:" + dimension; // unknown dimension ->
+                // Number
+                String actualItemType = channel.getAcceptedItemType();
+                if (!expectedItemType.equals(actualItemType)) {
+                    ChannelBuilder channelBuilder = callback.createChannelBuilder(channel.getUID(), channelTypeUID)
+                            .withAcceptedItemType(expectedItemType).withConfiguration(channel.getConfiguration());
+                    if (channel.getLabel() != null) {
+                        channelBuilder.withLabel(Objects.requireNonNull(channel.getLabel()));
```

Comment:
Perhaps this could be slightly simplified?
```suggestion
                    String label = channel.getLabel();
                    if (label != null) {
                        channelBuilder.withLabel(label);
```

also below.

Reply:
I guess it depends on the POV which version is simpler. I changed to according to your suggestion.